### PR TITLE
{tools} [intel/2016a] MariaDB 10.1.13 (REVIEW)

### PR DESCRIPTION
**Pull request by wpoely86**

### CODE REVIEW

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 2 out of 3 (2 easyconfigs in this PR)
node2159.delcatty.os - Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/2649728b1a04b8181d77aa6b4317341b for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2685.swalot.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/7a4956aa6281a18f9b31d38d8e8b4d75 for a full test report.

**Comment by boegel** [comment]
@wpoely86 hmm...

```

-- Looking for malloc_stats_print in jemalloc
-- Looking for malloc_stats_print in jemalloc - not found
-- Looking for malloc_stats_print in c
-- Looking for malloc_stats_print in c - not found
CMake Error at cmake/jemalloc.cmake:38 (MESSAGE):
  jemalloc is not found
```

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 1 out of 2 (2 easyconfigs in this PR)
node2159.delcatty.os - Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/e7216e52bd6863520a3f9472e64a3e9d for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2159.delcatty.os - Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/7c2e22f72233f21b2ce6d87e9ebd9ada for a full test report.

**Comment by boegel** [comment]
@wdpypere let me know if this is not WIP anymore...

**Comment by wpoely86** [comment]
@boegel it's ready.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2424.golett.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/ffd64ce705eadabdfce8e406d0e91269 for a full test report.

File: easybuild/easyconfigs/j/jemalloc/jemalloc-4.1.0-intel-2016a.eb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+sources = ['%(version)s.tar.gz']
+
+builddependencies = [('Autotools', '20150215')]
+
+preconfigopts = "./autogen.sh && "
+prebuildopts = "make dist && "
+
+sanity_check_paths = {
+    'files': ['bin/jeprof', 'lib/libjemalloc.a', 'lib/libjemalloc_pic.a', 'lib/libjemalloc.%s' % SHLIB_EXT,
+              'lib/libjemalloc.so.2', 'include/jemalloc/jemalloc.h'],
```
Comment (boegel) [inline easybuild/easyconfigs/j/jemalloc/jemalloc-4.1.0-intel-2016a.eb]: there's little point in using `SHLIB_EXT` above, but not for the `.so.2`...

File: easybuild/easyconfigs/j/jemalloc/jemalloc-4.1.0-intel-2016a.eb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    'dirs': ['share'],
+}
+
+modextrapaths = {'LD_PRELOAD': ['lib/libjemalloc.so.2']}
```
Comment (boegel) [inline easybuild/easyconfigs/j/jemalloc/jemalloc-4.1.0-intel-2016a.eb]: why don't we use the `.so` instead? any particular reason?

File: easybuild/easyconfigs/m/MariaDB/MariaDB-10.1.13-intel-2016a.eb [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+]
+
+dependencies = [
+    ('zlib', '1.2.8'),
+    ('ncurses', '6.0'),
+    ('jemalloc', '4.1.0'),
+    ('PCRE', '8.38'),
+    ('XZ', '5.2.2'),
+    ('Boost', '1.60.0', '-Python-2.7.11'),
+    ('libxml2', '2.9.3'),
```
Comment (boegel) [inline easybuild/easyconfigs/m/MariaDB/MariaDB-10.1.13-intel-2016a.eb]: @wpoely86 `libedit` dep was dropped?

File: easybuild/easyconfigs/m/MariaDB/MariaDB-10.1.13-intel-2016a.eb [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+]
+
+builddependencies = [('CMake', '3.4.3')]
+
+separate_build_dir = True
+
+configopts = "-DWITH_PCRE=system -DWITH_JEMALLOC=yes -DWITH_ZLIB=system -DMYSQL_MAINTAINER_MODE=ON"
```
Comment (boegel) [inline easybuild/easyconfigs/m/MariaDB/MariaDB-10.1.13-intel-2016a.eb]: `-DCURSES_LIBRARY=$EBROOTNCURSES/lib/libncurses.so -DCURSES_INCLUDE_PATH=$EBROOTNCURSES/include` no longer needed?

Comment (wpoely86) [inline easybuild/easyconfigs/m/MariaDB/MariaDB-10.1.13-intel-2016a.eb]: nope. It finds it without issue.

File: easybuild/easyconfigs/m/MariaDB/MariaDB-10.1.13-intel-2016a.eb (hunk 3 above)
Comment (wpoely86) [inline easybuild/easyconfigs/m/MariaDB/MariaDB-10.1.13-intel-2016a.eb]: yes, replaced by libreadline :laughing:

File: easybuild/easyconfigs/m/MariaDB/MariaDB-10.1.13-intel-2016a.eb (hunk 4 above)
Comment (boegel) [inline easybuild/easyconfigs/m/MariaDB/MariaDB-10.1.13-intel-2016a.eb]: well, ok then

File: easybuild/easyconfigs/m/MariaDB/MariaDB-10.1.13-intel-2016a.eb (hunk 3 above)
Comment (boegel) [inline easybuild/easyconfigs/m/MariaDB/MariaDB-10.1.13-intel-2016a.eb]: ah, ok

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 1 out of 2 (2 easyconfigs in this PR)
node2159.delcatty.os - Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/698d7e40e426d72d1c6183420a87e2ef for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2159.delcatty.os - Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/813fa0c906caaa7de91037d0a39646b8 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2091.delcatty.os - Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/002f07ff61da2141e06b389a554a3eaa for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 1 out of 2 (2 easyconfigs in this PR)
node2657.swalot.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/4b1d5b7f52e57653edc3af665be53fc7 for a full test report.

**Comment by wpoely86** [comment]
@boegel fs issue?
```
== 2016-04-21 21:51:35,028 build_log.py:152 ERROR Failed to read /user/data/gent/vsc400/vsc40023/EasyBuild/sources/m/MariaDB/mariadb-10.1.13.tar.gz: [Errno 5] Input/output error (at easybuild/tools/filetools.py:421 in calc_block_checksum)
```

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2566.golett.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/73d24fcaa31e1b43d21f9e98832299b9 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2618.swalot.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/28126496d413edb476b90d03c5875321 for a full test report.

**Comment by boegel** [comment]
@wpoely86 seems ok indeed, so going in \o/

File: easybuild/easyconfigs/j/jemalloc/jemalloc-4.1.0-intel-2016a.eb [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+preconfigopts = "./autogen.sh && "
+prebuildopts = "make dist && "
+
+sanity_check_paths = {
+    'files': ['bin/jeprof', 'lib/libjemalloc.a', 'lib/libjemalloc_pic.a', 'lib/libjemalloc.%s' % SHLIB_EXT,
+              'include/jemalloc/jemalloc.h'],
+    'dirs': ['share'],
+}
+
+modextrapaths = {'LD_PRELOAD': ['lib/libjemalloc.%s' % SHLIB_EXT]}
```
Comment (boegel) [inline easybuild/easyconfigs/j/jemalloc/jemalloc-4.1.0-intel-2016a.eb]: @wpoely86 shall we make this change in the existing `jemalloc` easyconfigs too?

**Pull request merged**